AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestMatrixStore.kt

```diff
@@ -140,6 +141,46 @@ internal class TestMatrixStore(
         return newTestMatrix
     }
 
+    /**
+     * Creates a [TestMatrix] to run the tests specified in [testTargets] list
+     * using the same configuration as [testMatrix]
+     */
+    suspend fun createTestMatrix(
+        testMatrix: TestMatrix,
+        testTargets: List<String>? = null
+    ): TestMatrix {
+        logger.trace {
+            "test matrix id: ${testMatrix.testMatrixId}"
+        }
+
+        val testSpecification = testMatrix.testSpecification.copy(
+            androidInstrumentationTest = testMatrix.testSpecification.androidInstrumentationTest?.copy(
+                testTargets = testTargets
+            )
+        )
+        val newTestMatrix = firebaseTestLabApi.createTestMatrix(
```

Comment:
sorry i missed this in the first release.

i'm a bit concerned about this because when we copy the previous test matrix, we might copy fields that we don't / shouldn't.

e.g. i'm not sure if client info is suitable for copying. AndroidxDev might stash unique ids in there :/. Could we also receive that parameter? 

also, the result path generation is different from `createNewTestMatrix`. Is it possible to read everything needed to call `createNewTestMatrix` and create the test matrix using `createNewTestMatrix`. That will also safe guard us for future changes.